### PR TITLE
[Intel GPU] fix matmul accuracy when offset > 0

### LINKED ISSUE #1656: XPU ops will return wrong results when handling specific offset inputs
### 🐛 Describe the bug

When running UTs for Flex Attention, I noticed that XPU returns results that differ from CPU.
Below is a simplified reproducer.
```
python reproducer.py
```

[reproducer-strides.tar.gz](https://github.com/user-attachments/files/20180934/reproducer-strides.tar.gz)

Additionally, CUDA returns correct results when reading inputs with the same stride.
Image

I found this issue while running UTs for the compiled Flex Attention. The compiled results are correct, so this issue seems to be specific to the eager mode.

---

@jianyizh helped us confirm that this issue is most likely caused by a specific offset.
The generation methods for the two tensors are as follows:
```python
B = 2
H = 4
S = 256
D = 64

q_s = ((H * S * D, S * D, D, 1), 997)
k_s = ((H * S * D, S * D, D, 1), 997)

dtype = torch.float16
device = 'xpu'

q1 = torch.randn((B * H * S * D * 2), dtype=dtype, device=device)
k1 = torch.randn((B * H * S * D * 2), dtype=dtype, device=device)

q_shape = (B, H, S // 2, D)
k_shape = (B, H, S, D)

requires_grad = True

def coerce_to_strides(val, shape, strides):
    strides, offset = strides
    val_max = [x * (y - 1) for x, y in zip(strides, shape)]
    assert sum(val_max) + offset < B * H * S * D * 2
    assert strides[-1] == 1
    return torch.as_strided(val, shape, strides, offset).requires_grad_(
        requires_grad
    )

q = coerce_to_strides(q1, q_shape, q_s)
k = coerce_to_strides(k1, k_shape, k_s)
```

### Versions

```
Collecting environment information...
PyTorch version: 2.8.0.dev20250512+xpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.4 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.10.17 | packaged by conda-forge | (main, Apr 10 2025, 22:19:12) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-5.15.50-051550-generic-x86_64-with-glibc2.35
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Address sizes:                   52 bits physical, 57 bits virtual
Byte Order:                      Little Endian
CPU(s):                          224
On-line CPU(s) list:             0-223
Vendor ID:                       GenuineIntel
Model name:                      Intel(R) Xeon(R) Platinum 8480+
CPU family:                      6
Model:                           143
Thread(s) per core:              2
Core(s) per socket:              56
Socket(s):                       2
Stepping:                        6
CPU max MHz:                     3800.0000
CPU min MHz:                     800.0000
BogoMIPS:                        4000.00
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 invpcid_single intel_ppin cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect avx_vnni avx512_bf16 wbnoinvd dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxldtrk pconfig arch_lbr avx512_fp16 flush_l1d arch_capabilities
Virtualization:                  VT-x
L1d cache:                       5.3 MiB (112 instances)
L1i cache:                       3.5 MiB (112 instances)
L2 cache:                        224 MiB (112 instances)
L3 cache:                        210 MiB (2 instances)
NUMA node(s):                    2
NUMA node0 CPU(s):               0-55,112-167
NUMA node1 CPU(s):               56-111,168-223
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Enhanced IBRS, IBPB conditional, RSB filling
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected

Versions of relevant libraries:
[pip3] numpy==2.1.2
[pip3] pytorch-triton-xpu==3.3.0+git0bcc8265
[pip3] torch==2.8.0.dev20250512+xpu
[pip3] torchaudio==2.6.0.dev20250512+xpu
[pip3] torchvision==0.22.0.dev20250512+xpu
[conda] numpy                     2.1.2                    pypi_0    pypi
[conda] pytorch-triton-xpu        3.3.0+git0bcc8265          pypi_0    pypi
[conda] torch                     2.8.0.dev20250512+xpu          pypi_0    pypi
[conda] torchaudio                2.6.0.dev20250512+xpu          pypi_0    pypi
[conda] torchvision               0.22.0.dev20250512+xpu          pypi_0    pypi
```

### PR DESCRIPTION
**Pull request by jianyizh**
This pr will make matmul tensors contiguous if they are not 64 byte alignment. oneDNN requires a minimal alignment of 64 https://uxlfoundation.github.io/oneDNN/dev_guide_c_and_cpp_apis.html#intel-r-processor-graphics-and-xe-architecture-graphics

Fixes https://github.com/intel/torch-xpu-ops/issues/1656

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168

### CODE REVIEW

**Review by liangan1 — APPROVED** [review]

**Review by guangyey — APPROVED** [review]
LGTM.
Could you please add a UT to `test/xpu/test_gemm.py`?

**Review by EikanWang — APPROVED** [review]
